### PR TITLE
FPGA: Add a new Reference Design sample: `niosv`

### PR DESCRIPTION
**Pull request by whitepau**
# Adding a New Sample(s)

## Description

This reference design shows a Nios® V softcore processor interacting with a FPGA IP produced with the Intel® oneAPI DPC++/C++ Compiler. It demonstrates considerations for how to structure embedded code to be aware of the registers in an FPGA IP produced with the Intel® oneAPI DPC++/C++ Compiler.

## Checklist
Administrative
- [X] Review sample design with the appropriate [Domain Expert](https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts): <insert Name Here>
- [X] If you have any new dependencies/binaries, inform the oneAPI Code Samples Project Manager

Code Development
- [X] Implement coding guidelines and ensure code quality. [see wiki for details](https://github.com/oneapi-src/oneAPI-samples/wiki/General-Code-Guidelines)
- [X] Adhere to readme template 
- [X] Enforce format via clang-format config file
- [X] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [ ] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [X] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
- [X] FPGA regtest 
* linux oneAPI: https://spetc.intel.com/testanalysis?trview=0&testRunIds=8764632&tapgsize=20
* linux niosV simulation: https://spetc.intel.com/testanalysis/?trview=0&testRunIds=8764623&tapgsize=20
- [X] FPGA regtest 
* windows oneAPI: https://spetc.intel.com/testanalysis?trview=0&testRunIds=8772540&tapgsize=20&tasort=testCasePath&tasortdir=asc
* windows niosV simulation: https://spetc.intel.com/testanalysis?trview=0&testRunIds=8764622&tapgsize=20&tasort=testCasePath&tasortdir=asc (reports as failure but if you look at the output it's correct)

Security and Legal
- [ ] OSPDT Approval (see Project Manager for assistance)
- [X] Compile using the following compiler flags and fix any warnings, the flags are: "/Wall -Wformat-security -Werror=format-security"
- [ ] Bandit Scans (Python only)
- [ ] Virus scan

Review
- [ ] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth(@tomlenth) and/or Project Manager
- [ ] Tested using Dev Cloud when applicable

### CODE REVIEW

**Comment by yuguen-intel** [comment]
@tomlenth @pmpeter1 Please review this new FPGA code sample

File: DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/niosv/kernels/simple_dma/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+# Use cmake -DFPGA_DEVICE=<board-support-package>:<board-variant> to choose a
+# different device. Here are a few device examples (this list is not
+# exhaustive):
+#   intel_s10sx_pac:pac_s10
+#   intel_s10sx_pac:pac_s10_usm
+#   intel_a10gx_pac:pac_a10
+# Note that depending on your installation, you may need to specify the full 
+# path to the board support package (BSP), this usually is in your install 
+# folder.
```
Comment (yuguen-intel) [inline DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/niosv/kernels/simple_dma/CMakeLists.txt:33]:
```suggestion
# Use cmake -DFPGA_DEVICE=<board-support-package>:<board-variant> to choose a
# different device. 
# Note that depending on your installation, you may need to specify the full 
# path to the board support package (BSP), this usually is in your install 
# folder.
```
Removing all mentions to the now unsupported PAC boards.

**Review by yuguen-intel — CHANGES_REQUESTED** [review]

**Review by yuguen-intel — APPROVED** [review]

File: DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/niosv/software/simple_dma_test/src/simple_dma_test.c [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+#include "../../../kernels/simple_dma/build/simple_dma.report.prj/include/register_map_offsets.hpp"
+
+// In bytes, must be a multiple of 4.  Keep it a small number to shorten the
+// simulation time and do not exceed the 1MB memory size (remember this code is
+// in there too). Once the DMA gets going this buffer will fly by fast so
+// setting it too high mostly affects the memory initialization and the
+// correctness check at the end once the DMA gets going this buffer will fly by
+// fast so setting it too high mostly affects the memory initialization and the
+// correctness check at the end.
+#define BUFFER_LENGTH 1024
```
Comment (pmpeter1) [inline DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/niosv/software/simple_dma_test/src/simple_dma_test.c:39]: When using SYCL/C++ it's prefered to not use #define if their are more typesafe ways of solving the problem.   The usual solution is to use constexpr to define correct types of expressions and values.

**Review by pmpeter1 — CHANGES_REQUESTED** [review]

File: DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/niosv/software/simple_dma_test/src/simple_dma_test.c (hunk 2 above)
Comment (yuguen-intel) [reply DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/niosv/software/simple_dma_test/src/simple_dma_test.c:39]: This is not SYCL code, this is C code for the softprocessor that will be used in combination with the SYCL code compiled in `simple_dma.cpp`.
`simple_dma.cpp` is the SYCL code that is given to our SYCL/C++ oneapi compiler. I don't think file is compiled by our compiler, and I think that this is restricted to C, which does not support `constexpr` I believe

**Review by ishaghosh27 — APPROVED** [review]

File: DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/niosv/software/simple_dma_test/src/simple_dma_test.c (hunk 2 above)
Comment (whitepau) [reply DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/niosv/software/simple_dma_test/src/simple_dma_test.c:39]: @yuguen-intel  is correct.

**Review by yuguen-intel — APPROVED** [review]

**Pull request merged**